libs/agno/agno/vectordb/couchbase/couchbase.py

```diff
@@ -0,0 +1,483 @@
+import time
+from datetime import timedelta
+from typing import Any, Dict, List, Optional, Union
+
+from agno.document import Document
+from agno.embedder import Embedder
+from agno.embedder.openai import OpenAIEmbedder
+from agno.utils.log import logger
+from agno.vectordb.base import VectorDb
+
+try:
+    from hashlib import md5
+
+except ImportError:
+    raise ImportError("`hashlib` not installed. Please install using `pip install hashlib`")
+try:
+    from couchbase.cluster import Cluster
+    from couchbase.collection import Collection
+    from couchbase.exceptions import (
+        BucketDoesNotExistException,
+        DocumentNotFoundException,
+        ScopeAlreadyExistsException,
+        SearchIndexNotFoundException,
+    )
+    from couchbase.management.search import ScopeSearchIndexManager, SearchIndex, SearchIndexManager
+    from couchbase.n1ql import QueryScanConsistency
+    from couchbase.options import ClusterOptions, QueryOptions, SearchOptions
+    from couchbase.result import SearchResult
+    from couchbase.scope import Scope
+    from couchbase.search import SearchRequest
+    from couchbase.vector_search import VectorQuery, VectorSearch
+except ImportError:
+    raise ImportError("`couchbase` not installed. Please install using `pip install couchbase`")
+
+
+class CouchbaseFTS(VectorDb):
```

Comment:
```suggestion
class Couchbase(VectorDb):
```
What do you think about this change?

Reply:
While the Couchbase name is more concise, I believe retaining CouchbaseSearch (changed from CouchbaseFTS in the latest commit) is preferable as we're planning to support vector search through multiple Couchbase services (both FTS and the indexing service). The more specific naming clearly indicates which implementation is being used and will help avoid confusion when we add additional Couchbase vector search capabilities in the future.